### ISSUE TITLE
Can't switch to buffers by name

### ISSUE THREAD

**Issue by neutralinsomniac** [poljar/weechat-matrix#36]
I'm used to using the `/buffer <name>` command with the weechat lua script to switch to particular buffers (since I have well over 100 of them). I noticed that while the weechat shortname for my buffers is correct, weechat is using the room ID as the key to complete the /buffer command (server.!roomid:server.com). Is there a weechat setting that will make the /buffer command evaluate short names for completion/buffer switching or is there some other way I should be switching directly to buffers with your script?

**Comment by neutralinsomniac** [comment]
Awesome, thank you!

**Comment by ara4n** [comment]
i'm also being bitten by this (even though it's closed) - i got too used to the lua weechat script letting me type `/buffer HQ` to jump to matrix HQ and other such partial matches.  I think it might need the command overriding with `hook_command_run` as proposed? @poljar - it might be worth reopening perhaps?  Or should I use go.py instead?

**Comment by poljar** [comment]
Yes we can reopen this, there probably is a way to implement some sort of functionality `/buffer <buffer-name>` functionality.

In the meantime I do recommend go.py since it has nice fuzzy matching.

**Comment by poljar** [comment]
I just pushed 01b5d5c1c5faca9bfcf7db97ee2eca4ca0102afe to master that implements this. If someone could confirm that it works as expected, that would be great.

**Comment by neutralinsomniac** [comment]
It works! Not to sound ungrateful, but the lua script also did tab-completion for the /buffer command. Is that easy to add? :)

**Comment by poljar** [comment]
The Lua script just used the default completion that Weechat already provides. The problem is that Weechat completes the full name and that is again the room id for buffers that belong to this script.

I don't think it should be hard to add, overriding completion items is supported by Weechat. Will check it out.

**Comment by poljar** [comment]
I added https://github.com/poljar/weechat-matrix/commit/75b2b85626a94bed3d5b1db928c52ee4c83ea26a that introduces a new completion template. If you add `%(matrix_rooms)` to the `weechat.completion.default_template` it will complete the short buffer name of matrix rooms.

Another setting called `matrix.look.human_buffer_names` was added so the full buffer name can consist of the server name + the short name instead of the matrix room id. A user was complaining that he couldn't switch buffers by name in a Weechat relay client, since we can't override commands for relay clients this was necessary to do.

Note: that setting can't be used by default, it will work well only if the `logger.file.mask` is set not to use the buffer name (e.g. use the $channel variable instead).

I will close this soon if nobody has any further objections.
